libraries/ESP8266HTTPClient/examples/BasicHttpsClient/BasicHttpsClient.ino

```diff
@@ -45,32 +45,36 @@ void loop() {
 
     client->setFingerprint(fingerprint);
 
-    HTTPClient https;
-
-    Serial.print("[HTTPS] begin...\n");
-    if (https.begin(*client, "https://jigsaw.w3.org/HTTP/connection.html")) {  // HTTPS
-
-      Serial.print("[HTTPS] GET...\n");
-      // start connection and send HTTP header
-      int httpCode = https.GET();
-
-      // httpCode will be negative on error
-      if (httpCode > 0) {
-        // HTTP header has been send and Server response header has been handled
-        Serial.printf("[HTTPS] GET... code: %d\n", httpCode);
-
-        // file found at server
-        if (httpCode == HTTP_CODE_OK || httpCode == HTTP_CODE_MOVED_PERMANENTLY) {
-          String payload = https.getString();
-          Serial.println(payload);
+    /* in order to ensure the client object lives the entire time of the HTTPClient
+       the HTTPClient is in its own nested code block */
+    {
```

Comment:
HTTPClient doesn't need to be in its own code block, it just needs to be instantiated _after_ client. Destructors are called in reverse order, which ensures https gets destroyed first and client second.
I suggest removing the code block, and updating the comment to explain this.

Reply:
I don't agree. It's way too dangerous to rely on order of code lines. Took me two days to debug that. It should be totally explicit, IMHO.